### ISSUE TITLE
Add support for modifing `Value` and `Object`

### ISSUE THREAD

**Issue by PoiScript** [khonsulabs/justjson#6]
In my use case, we need to rewrite some json files, like conditional adding or removing some fields from objects.

Currently, we're using `serde_json` and `RawValue` to do the job. It does work great.
But we would like to try out `justjson` to see if we can make it even more efficient. :)

However, `justjson` doesn't export `Entry` struct to public. And both `Value` and `Object` is immutable.
Do you have any idea supporting it?

**Comment by ecton** [comment]
Thank you for the suggestion! The lack of exporting Entry was an oversight, my bad!

Value is only immutable in that you must first destructure the Value type to know what's inside of it. Once you have an `Object` or a Sequence's Vec, you can modify it. `Object` implements DerefMut to its underlying `Vec<Entry>`, so if I had properly exported Entry, you would be able to push entries in addition to modify the entries.

Here's a snippet from a unit test where modification is tested:

```rust
#[test]
fn cow() {
    let mut value =
        Value::from_json_bytes(br#"{"a":1,"b":true,"c":"hello","d":[],"e":{}}"#).unwrap();
    let Value::Object(root) = &mut value else { unreachable!() };
    root[0].key = JsonString::from("newa");
    let Value::Array(d_array) = &mut root[3].value else { unreachable!() };
    d_array.push(Value::Null);
    let generated = value.to_json();
    assert_eq!(
        generated,
        r#"{"newa":1,"b":true,"c":"hello","d":[null],"e":{}}"#
    );
}
```

I'll put together a better example soon, but more importantly, I'll get Entry exported in the next release! Additionally, the suggestion in issue #5 could end up helping with the usability of mutating Values.

**Comment by ecton** [comment]
I've released [v0.2.2](https://github.com/khonsulabs/justjson/releases/tag/v0.2.2) with some major usability improvements as well as the export of the `Entry` type. I also have a new doc example showing [creating an object](https://khonsulabs.github.io/justjson/main/justjson/struct.Object.html#modifying-an-object) using push.

I'm sure there are still some rough edges when editing values -- please don't hesitate to ask for further improvements! I haven't personally used this for value modification in a project yet, I've only used it to parse so far.
